consul/resource_service_defaults_config_entry.go

```diff
@@ -0,0 +1,557 @@
+			if err != nil {
+				return nil, err
+			}
+			if formattedValue != nil {
+				formattedMap[formattedKey] = formattedValue
+			}
+		}
+		return formattedMap, nil
+	} else if isSlice(config) {
+		var newSlice []interface{}
+		listValue := config.([]interface{})
+		for _, elem := range listValue {
+			newElem, err := formatKeys(elem, formatKey)
+			if err != nil {
+				return nil, err
+			}
+			newSlice = append(newSlice, newElem)
+		}
+		return newSlice, nil
+	} else if isStruct(config) {
+		var modifiedStruct map[string]interface{}
+		jsonValue, err := json.Marshal(config)
+		if err != nil {
+			return nil, err
+		}
+		err = json.Unmarshal(jsonValue, &modifiedStruct)
+		if err != nil {
+			return nil, err
+		}
+		formattedStructKeys, err := formatKeys(modifiedStruct, formatKey)
+		if err != nil {
+			return nil, err
+		}
+		return formattedStructKeys, nil
+	} else if isSetSchema(config) {
+		valueList := config.(*schema.Set).List()
+		if len(valueList) > 0 {
+			formattedSetValue, err := formatKeys(valueList[0], formatKey)
+			if err != nil {
+				return nil, err
+			}
+			return formattedSetValue, nil
+		}
+		return nil, nil
+	}
+	return config, nil
+}
+
+func resourceConsulServiceDefaultsConfigEntryUpdate(d *schema.ResourceData, meta interface{}) error {
+	client, qOpts, wOpts := getClient(d, meta)
+	configEntries := client.ConfigEntries()
+
+	name := d.Get("name").(string)
+
+	configMap := make(map[string]interface{})
+	configMap["kind"] = "service-defaults"
+
+	configMap["name"] = name
+
+	kind := configMap["kind"].(string)
+	err := d.Set("kind", kind)
+
+	if err != nil {
+		return err
+	}
+
+	fixQOptsForServiceDefaultsConfigEntry(name, kind, qOpts)
+
+	var attributes []string
+
+	for key, _ := range serviceDefaultsConfigEntrySchema {
+		attributes = append(attributes, key)
+	}
+
+	for _, attribute := range attributes {
+		configMap[attribute] = d.Get(attribute)
+	}
+
+	formattedMap, err := formatKeys(configMap, formatKey)
+	if err != nil {
+		return err
+	}
+
+	configEntry, err := makeServiceDefaultsConfigEntry(kind, name, formattedMap.(map[string]interface{}), wOpts.Namespace, wOpts.Partition)
+	if err != nil {
+		return err
+	}
+
+	if _, _, err := configEntries.Set(configEntry, wOpts); err != nil {
+		return fmt.Errorf("failed to set '%s' config entry: %v", name, err)
+	}
+	_, _, err = configEntries.Get(configEntry.GetKind(), configEntry.GetName(), qOpts)
+	if err != nil {
+		if strings.Contains(err.Error(), "Unexpected response code: 404") {
+			return fmt.Errorf(`failed to read config entry after setting it.
+This may happen when some attributes have an unexpected value.
+Read the documentation at https://www.consul.io/docs/agent/config-entries/%s.html
+to see what values are expected`, configEntry.GetKind())
+		}
+		return fmt.Errorf("failed to read config entry: %v", err)
+	}
+
+	d.SetId(fmt.Sprintf("%s-%s", kind, name))
+	return resourceConsulServiceDefaultsConfigEntryRead(d, meta)
```

Comment:
We seem to do a `Get` on the same resource for setting the `Id` attribute. Any reason to do it again here?

Reply:
First we call the update API then we do a get to confirm that resource is created with that id. If get gives an error that means update failed.

Reply:
This similar thing is done for config_entry resource as well.